workers/unity/Packages/com.improbable.gdk.core/Utility/EntityBuilder.cs

```diff
@@ -0,0 +1,211 @@
+        }
+
+        public EntityBuilder SetPersistence(bool persistence)
+        {
+            if (persistence)
+            {
+                var schemaData = new SchemaComponentData(WellKnownComponents.Persistence.ComponentId);
+                entity.Add(new ComponentData(schemaData));
+            }
+            else
+            {
+                if (entity.Get(WellKnownComponents.Persistence.ComponentId).HasValue)
+                {
+                    entity.Remove(WellKnownComponents.Persistence.ComponentId);
+                }
+            }
+
+            return this;
+        }
+
+        public EntityBuilder AddMetadata(string metadata, string writeAccess)
+        {
+            var schemaData = new SchemaComponentData(WellKnownComponents.Metadata.ComponentId);
+            var fields = schemaData.GetFields();
+            fields.AddString(1, metadata);
+
+            return AddComponent(new ComponentData(schemaData), writeAccess);
+        }
+
+        public EntityBuilder SetReadAcl(string attribute, params string[] attributes)
+        {
+            acl.AddReadAccess(attribute);
+            foreach (var attr in attributes)
+            {
+                acl.AddReadAccess(attr);
+            }
+
+            return this;
+        }
+
+        public EntityBuilder SetReadAcl(List<string> attributes)
+        {
+            foreach (var attribute in attributes)
+            {
+                acl.AddReadAccess(attribute);
+            }
+
+            return this;
+        }
+
+        public EntityBuilder SetEntityAclComponentWriteAccess(string attribute)
+        {
+            acl.SetComponentWriteAccess(WellKnownComponents.EntityAcl.ComponentId, attribute);
+            return this;
+        }
+
+        public Entity Build()
+        {
+            // Build and add the entity acl
+            entity.Add(acl.Build());
+            CheckRequiredComponents();
+            return entity;
+        }
+
+        private void CheckRequiredComponents()
+        {
+            foreach (var requiredComponent in WellKnownComponents.RequiredComponents)
+            {
+                if (!componentsAdded.Contains(requiredComponent.ComponentId))
+                {
+                    throw new InvalidEntityException(
+                        $"Entity is invalid. Missing required component: {requiredComponent.ComponentId}");
+                }
+            }
+        }
+
+        private class InvalidEntityException : Exception
+        {
+            public InvalidEntityException(string message) : base(message)
+            {
+            }
+        }
+
+        private struct WellKnownComponent
+        {
+            public uint ComponentId;
+            public string Name;
+        }
+
+        private static class WellKnownComponents
+        {
+            public static readonly WellKnownComponent Position = new WellKnownComponent
+            {
+                ComponentId = 54,
```

Comment:
I recommend to use the named consts for these values too

Reply:
Yeah - having them in the static WKC was what I was planning to do to satisfy that. I.e. - instead of 

```
            public static readonly WellKnownComponent Position = new WellKnownComponent
            {
                ComponentId = PositionComponentId,
```

But I can refactor into consts